frontend/public/components/namespace.jsx

```diff
@@ -831,43 +837,44 @@ export const ProjectsPage = connectToFlags(
 /** @type {React.SFC<{namespace: K8sResourceKind}>} */
 export const PullSecret = (props) => {
   const [isLoading, setIsLoading] = React.useState(true);
-  const [data, setData] = React.useState(undefined);
+  const [data, setData] = React.useState([]);
   const { t } = useTranslation();
+  const { namespace } = props;
 
   React.useEffect(() => {
-    k8sGet(SecretModel, null, props.namespace.metadata.name, {
-      queryParams: { fieldSelector: 'type=kubernetes.io/dockerconfigjson' },
-    })
-      .then((pullSecrets) => {
-        setIsLoading(false);
-        setData(_.get(pullSecrets, 'items[0]'));
+    k8sGet(ServiceAccountModel, 'default', namespace.metadata.name, {})
+      .then((serviceAccount) => {
+        setData(serviceAccount.imagePullSecrets ?? []);
       })
       .catch((error) => {
-        setIsLoading(false);
         setData(undefined);
-        // A 404 just means that no pull secrets exist
-        if (error.status !== 404) {
-          throw error;
-        }
+        throw error;
+      })
+      .finally(() => {
+        setIsLoading(false);
       });
-  }, [props.namespace.metadata.name]);
+  }, [namespace.metadata.name]);
 
-  if (isLoading) {
-    return <LoadingInline />;
-  }
-  const modal = () =>
-    configureNamespacePullSecretModal({ namespace: props.namespace, pullSecret: data });
+  const modal = () => configureNamespacePullSecretModal({ namespace, pullSecret: undefined });
+
+  const secrets =
+    data.length > 0 ? (
```

Comment:
Will this be a runtime error if the service account request fails? We set data to `undefined` in the catch block.